lib/template-mixins.js

```diff
@@ -95,12 +95,13 @@ module.exports = function (t, fields, options) {
         });
     }
 
-    function optionGroup(key) {
-        return {
+    function optionGroup(key, extension) {
+        extension = extension || {};
+        return _.extend(extension, {
             'key': key,
             'error': this.errors && this.errors[key],
-            'legend': fields[key].legend && fields[key].legend.value,
-            'options': _.map(fields[key].options, function (obj) {
+            'legend': fields[key] && fields[key].legend && fields[key].legend.value,
```

Comment:
simple test such as the following will fail:
https://github.com/UKHomeOffice/passports-template-mixins/blob/master/test/spec.index.js#L55

So I mirrored this defensive approach: https://github.com/UKHomeOffice/passports-template-mixins/blob/master/lib/template-mixins.js#L82